test/api/search_controller_test.rb

```diff
@@ -20,14 +20,45 @@ class SearchControllerTest < ActionDispatch::IntegrationTest
   test 'should get all search facets' do
     puts "#{File.basename(__FILE__)}: #{self.method_name}"
 
-    facet_count = SearchFacet.count
+    facet_count = SearchFacet.visible.count
     execute_http_request(:get, api_v1_search_facets_path)
     assert_response :success
     assert json.size == facet_count, "Did not find correct number of search facets, expected #{facet_count} but found #{json.size}"
 
     puts "#{File.basename(__FILE__)}: #{self.method_name} successful!"
   end
 
+  test 'should get visible search facets' do
+    puts "#{File.basename(__FILE__)}: #{self.method_name}"
+
+    # make one random facet not visible
+    invisible_facet = SearchFacet.all.sample
+    invisible_facet.update(visible: false)
```

Comment:
```suggestion
    invisible_facet.update!(visible: false)
```
To avoid case where the update silently fails, and then the test isn't testing what we want

Reply:
For belt-and-suspenders, I would also add an `assert SearchFacet.visible.count == SearchFacet.count - 1`

Reply:
Great point.

Reply:
a30a464